lib/modules/manager/fleet/extract.ts

```diff
@@ -0,0 +1,107 @@
+import { loadAll } from 'js-yaml';
+import { regEx } from '../../../util/regex';
+import { GitTagsDatasource } from '../../datasource/git-tags';
+import { HelmDatasource } from '../../datasource/helm';
+import { checkIfStringIsPath } from '../terraform/util';
+import type { PackageDependency, PackageFile } from '../types';
+import type { FleetFile, FleetFileHelm, GitRepo } from './types';
+
+function extractGitRepo(doc: GitRepo): PackageDependency {
+  const tempDep: PackageDependency = {
+    depType: 'git_repo',
+    datasource: GitTagsDatasource.id,
+  };
+
+  const repo = doc.spec?.repo;
+  if (!repo) {
+    return {
+      ...tempDep,
+      skipReason: 'missing-depname',
+    };
+  }
+  tempDep.sourceUrl = repo;
+  tempDep.depName = repo;
+
+  const currentValue = doc.spec.revision;
+  if (!currentValue) {
+    return {
+      ...tempDep,
+      skipReason: 'no-version',
+    };
+  }
+
+  return {
+    ...tempDep,
+    currentValue,
+  };
+}
+
+function extractFleetFile(doc: FleetFileHelm): PackageDependency {
+  const tempDep: PackageDependency = {
+    depType: 'fleet',
+    datasource: HelmDatasource.id,
+  };
+
+  if (!doc.chart) {
+    return {
+      ...tempDep,
+      skipReason: 'missing-depname',
+    };
+  }
+  tempDep.depName = doc.chart;
+  tempDep.packageName = doc.chart;
+
+  if (!doc.repo) {
+    if (checkIfStringIsPath(doc.chart)) {
+      return {
+        ...tempDep,
+        skipReason: 'local-chart',
+      };
+    }
+    return {
+      ...tempDep,
+      skipReason: 'no-repository',
+    };
+  }
+  tempDep.registryUrls = [doc.repo];
+
+  const currentValue = doc.version;
+  if (!doc.version) {
+    return {
+      ...tempDep,
+      skipReason: 'no-version',
+    };
+  }
+  tempDep.currentValue = currentValue;
+
+  return tempDep;
+}
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string
+): PackageFile | null {
+  if (!content) {
+    return null;
+  }
+  const deps: PackageDependency[] = [];
+
+  if (regEx('fleet.ya?ml').test(packageFile)) {
```

Comment:
is there some documentation about this filename behavior?

Reply:
is there a possibility to get both `fleet.yaml` and `myGitRepoManifests.yaml` in the same repo ?
As I see you support either that or that, but not both in the same run

Reply:
In case both can be in the same repo, maybe better to update the readme.md file from:
```
  "fleet": {
    "fileMatch": ["myGitRepoManifests\\.yaml"]
  }
```
to:
```
  "fleet": {
    "fileMatch": [
        "'(^|/)fleet.ya?ml",
        "myGitRepoManifests\\.yaml"
    ]
  }
```